cooperator_portal/controllers/portal.py

```diff
@@ -90,17 +82,25 @@ def account(self, redirect=None, **post):
         partner = request.env.user.partner_id
 
         res = super().account(redirect, **post)
-        if not res.qcontext.get("error"):
-            partner_bank = request.env["res.partner.bank"]
+        if (
+            post
+            and request.httprequest.method == "POST"
+            and not res.qcontext.get("error")
+        ):
             iban = post.get("iban")
             if iban:
                 if partner.bank_ids:
-                    bank_account = partner.bank_ids[0]
-                    bank_account.acc_number = iban
+                    # update existing bank account
+                    partner.bank_ids[0].acc_number = iban
                 else:
-                    partner_bank.sudo().create(
-                        {"partner_id": partner.id, "acc_number": iban}
-                    )
+                    # create a new bank account
+                    partner.bank_ids = [
+                        (0, 0, {"partner_id": partner.id, "acc_number": iban})
+                    ]
+            else:
+                if partner.bank_ids:
+                    # delete the existing bank account
+                    partner.bank_ids = [(2, partner.bank_ids[0].id, 0)]
```

Comment:
What if the partner has several bank account ? Will it remove all of them ?

Reply:
this code follows the limitations of the module, which assumes that there is only one or zero bank accounts. in case there are multiple ones, it will delete the first one (the one that is displayed in the form).